### PR TITLE
Promisification

### PR DESCRIPTION
**Pull request by millette**
First stab at fixing #1.

Not sure about the new options argument (to receive the complete res or headers in the response). Happy to fix it however you see fit.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "gh-user",
-  "version": "1.0.0",
+  "version": "2.0.0",
```
Comment (sindresorhus) [inline package.json]: Generally don't bump the version in PRs. Owners have their own release process ;)

File: package.json [hunk 2]
```diff
@@ -26,9 +26,10 @@
     "info"
   ],
   "dependencies": {
-    "gh-got": "^1.0.3"
+    "gh-got": "^2.0.0",
+    "pinkie-promise": "^1.0.0"
   },
   "devDependencies": {
-    "ava": "0.0.4"
+    "ava": "0.1.0"
```
Comment (sindresorhus) [inline package.json]: `"ava": "*"`

File: index.js [hunk 3]
```diff
@@ -1,36 +1,36 @@
 'use strict';
-var ghGot = require('gh-got');
+var ghGot = require('gh-got'), Promise = require('pinkie-promise');
```
Comment (sindresorhus) [inline index.js]:
```js
var ghGot = require('gh-got');
var Promise = require('pinkie-promise');
```

**Comment by sindresorhus** [comment]
> Not sure about the new options argument (to receive the complete res or headers in the response)

I don't really see the point of that. Why would you need the headers? I'd rather just keep the output as it was without complicating it.

File: readme.md [hunk 4]
```diff
@@ -24,22 +24,33 @@ ghUser('sindresorhus', function (err, user) {
 
 ## API
 
-### ghUser(username, [token], callback)
+### ghUser(username, [options], [token])
 
 #### username
 
-*Required*  
+*Required*
```
Comment (sindresorhus) [inline readme.md]: Don't remove

File: readme.md [hunk 5]
```diff
@@ -24,22 +24,33 @@ ghUser('sindresorhus', function (err, user) {
...
+
+Type: `object`
+
+Passing an object with keys `res` or `headers` set to true will return
+an object with the keys `user` for the user object and either `res` or
+`headers` for the full res response or just the headers respectively.
+
 #### token
 
-Type: `string`
```
Comment (sindresorhus) [inline readme.md]: Don't remove

File: test.js [hunk 6]
```diff
@@ -1,12 +1,42 @@
...
+	return ghUser('sindresorhus').then(function (user) {
+		t.is(user.login, 'sindresorhus');
+	});
+});
+
+test('user exists, got headers', function (t) {
+  t.plan(2);
+	return ghUser('sindresorhus', {headers:true}).then(function (data) {
+		t.is(data.user.login, 'sindresorhus');
+    t.is(data.headers.status, '200 OK');
```
Comment (sindresorhus) [inline test.js]: indent

**Comment by sindresorhus** [comment]
Noticed some inconsistent indentation. Make sure you've used tab indentation.

**Comment by millette** [comment]
The headers are handy for the ratelimit, that's about it. But the limits are very generous (with a token), so it might not be an issue. I can remove all that, you're probably right. I'm coming the the search api and the limit are a bit stricter there.

**Comment by millette** [comment]
I guess I should now update the code (and watch the indent) given the notes above?

**Comment by sindresorhus** [comment]
If you need the rate limit info, you could expose it as a hidden property on the returned object:

```
Object.defineProperty(ret, 'rateLimit', {
  value: headers.rateLimit
});
```

**Comment by sindresorhus** [comment]
Yup

File: readme.md (hunk 5 above)
Comment (millette) [inline readme.md]: I didn't touch that line... Pushing other updates now.

Comment (sindresorhus) [inline readme.md]: You removed the whitespace that is used to force a linebreak.

File: readme.md [hunk 7]
```diff
@@ -35,11 +35,11 @@ GitHub username.
 
 #### token
 
-Type: `string`
```
Comment (sindresorhus) [inline readme.md]: Don't remove

File: test.js [hunk 8]
```diff
@@ -1,12 +1,27 @@
 'use strict';
-var test = require('ava');
 var ghUser = require('./');
+var test = require('ava');
```
Comment (sindresorhus) [inline test.js]: Don't move this.

File: readme.md [hunk 9]
```diff
@@ -47,7 +47,6 @@ Type: `object`
 
 Various [user info](https://developer.github.com/v3/users/#response).
 
-
```
Comment (sindresorhus) [inline readme.md]: Don't remove

File: index.js [hunk 10]
```diff
@@ -1,36 +1,20 @@
...
-
-		// remove deprecated props
-		delete data.gravatar_id;
-		delete data.bio;
-
-		cb(null, data);
+	}).then(function(res) {
+		delete res.body.gravatar_id;
+		delete res.body.bio;
+    return res.body;
```
Comment (sindresorhus) [inline index.js]: wrong indent

File: readme.md (hunk 7 above)
Comment (millette) [inline readme.md]: Again, not sure what the problem is with this line.

Comment (millette) [inline readme.md]: The two spaces at the end of the line?
Do you want the same space added at the end of the type: string above for the username argument?

Comment (sindresorhus) [inline readme.md]: As explained earlier, don't change it. The two spaces have semantic value in Markdown. It's a linebreak.

Comment (millette) [inline readme.md]: Our messages crossed. The linebreaks have been restored. In your original readme, some type: `string` have a linebreak, some don't. I was careful to do it exactly like yours. I was used to having my editor change a bunch of stuff on save, now that is disabled.

**Comment by millette** [comment]
I also added the hidden rate limit property in [another branch](https://github.com/millette/gh-user/commit/dd6d0247fc157c14e819de6e3c0d219ad2e312ac) - let me know if I should merge it with this or submit another pull request.

Thanks for the guidance, I appreciate it.

**Comment by sindresorhus** [comment]
Thanks :)

> let me know if I should merge it with this or submit another pull request.

Another PR.

**Pull request closed without merging**